### PR TITLE
CTONET-880 - make sure MCO does not contradict with sysctl config files

### PR DESCRIPTION
**Pull request by isaacdorfman**

### CODE REVIEW

File: test-network-function/generic/suite.go [hunk 1]
```diff
@@ -510,6 +586,21 @@ func testBootParams(context *interactive.Context, podName, podNamespace string,
...
 }
 
+func testSysctlConfigs(context *interactive.Context, podName, podNamespace string) {
+	ginkgo.It(fmt.Sprintf("Testing sysctl config files for the pod's node %s/%s", podNamespace, podName), func() {
+		nodeName := getPodNodeName(context, podName, podNamespace)
+		combinedSysctlSettings := getSysctlConfigArgs(context, nodeName)
+		mcName := getMcName(context, nodeName)
+		mcKernelArgumentsMap := getMcKernelArguments(context, mcName)
+
+		for key, sysctlConfigVal := range combinedSysctlSettings {
```
Comment (crwr45) [inline test-network-function/generic/suite.go:596]: I don't know enough about how the two different sets of values interact to know if this is correct but currently this is not asserting in all cases.

Currently if a value is:
1. present in both maps and equal -> assertion passes
2. present in both maps but unequal -> assertion fails
3. present only in `sysctlConfigVal` -> triggers no assertion, therefore passes
4. present only in `mcKernelArgumentsMap` -> triggers no assertion, therefore passes

@ryandgoulding do you know if the current behaviour is correct?

File: pkg/tnf/handlers/readremotefile/readremotefile.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+// ReadRemoteFile holds information regarding remote file contents at a specified node and path.
+type ReadRemoteFile struct {
+	remoteFileContents string // Output variable that stores the remote file contents
+	result             int
+	timeout            time.Duration
+	args               []string
+}
+
+// NewReadRemoteFile creates a ReadBootConfig tnf.Test.
```
Comment (crwr45) [inline pkg/tnf/handlers/readremotefile/readremotefile.go:39]: update comment

**Review by crwr45 — CHANGES_REQUESTED** [review]

File: test-network-function/generic/suite.go (hunk 1 above)
Comment (isaacdorfman) [reply test-network-function/generic/suite.go:582]: You understand it correctly.
This is the way it supposed to behave.

File: test-network-function/generic/suite.go [hunk 3]
```diff
@@ -420,13 +427,17 @@ func getMcName(context *interactive.Context, nodeName string) string {
 	return mcNameTester.GetMcName()
 }
 
+func runAndValidateTest(test *tnf.Test) {
```
[the remark this answers is not in the document]
Comment (isaacdorfman) [reply test-network-function/generic/suite.go]: Done

File: test-network-function/generic/suite.go [hunk 4]
```diff
@@ -491,6 +502,71 @@ func getGrubKernelArgs(context *interactive.Context, nodeName string) map[string
...
+	for _, line := range splitConfig {
+		if line == "" {
+			continue
+		}
+
+		if strings.HasPrefix(line, "#") || strings.HasPrefix(line, ";") {
+			continue
+		}
+
+		keyValRegexp := regexp.MustCompile("(\" \")*\\S*(\" \")*\\S*(\" \")*")
```
[the remark this answers is not in the document]
Comment (isaacdorfman) [reply test-network-function/generic/suite.go]: I changed the regex.

**Comment by ryandgoulding** [comment]
This fails smoke tests, and will not work on minikube.  Please add logic to skip for minikube environment.

**Pull request merged**